### ISSUE TITLE
expand_path_into_runfiles does not handle $locations

### ISSUE THREAD

**Issue by mattem** [bazelbuild/rules_nodejs#461]
# 🐞 bug report

### Affected Rule

The issue is caused by the rule: `nodejs_binary` (kinda)

### Is this a regression?

No

### Description

When passing data to `nodejs_binary` via the `templated_args` and using `$locations` expansion, only the first file path gets the workspace name added as the prefix and path resolved.

This is our use case... We have a JS file that uploads files to S3 and takes a flat list of files as its arguments (plus some other things, not relevant here). We've wrapped this in a macro, so we can do something like:

```
s3_push(
    name = "s3",
    files = [
        ":bundle",
       ...others omitted...
    ],
)
```
(Where `:bundle` points to the output of `rollup_bundle` in this example)
In the macro, it appends the `locations` expansion to each target:

```
file_paths = [("$(locations %s) " % file) for file in files]
```

After expansion, only the first file has the correct path. Looking at the `expand_path_into_runfiles` function, it doesn't handle the space separated strings that `expand_location` appears to return.

**Comment by alexeagle** [comment]
Actually we should delete expand_path_into_runfiles as part of @manekinekko work

**Comment by mattem** [comment]
We are looking at publishing our rules_aws rules to Github, but this issue currently blocks that as it requires patching rules_nodejs (which we do internally). Would you accept a PR for a fix while other work is going on?

**Comment by gregmagolan** [comment]
This was resolved with #727. nodejs_binary calls `expand_location_into_runfiles` which handles $locations.
